### PR TITLE
Add prettier linter for stimulusjs files

### LINKED ISSUE #9445: Add prettier to our stack
### What we should change and why (this is tech debt)
As we will more and more use StimulusJS and add controllers, it's a good idea to all have the same formatter in order to have a clean git history. 
https://prettier.io/ is the tool for that

Question is:
 - should we add a git hook? https://prettier.io/docs/en/install.html#git-hooks
 - should we use a gh action? https://github.com/marketplace/actions/actionsx-prettier

### Context
This came up in PR  https://github.com/openfoodfoundation/openfoodnetwork/pull/9444#issuecomment-1193722428

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

Closes #9445 
This PR basically: 
 - [x] adds prettier bin to our stack
 - [x] configure all ignored files (but not the ones related to stimulusjs)
 - [ ] ~adds a gh action that should run prettier through reviewdog~
 - [x] adds a git pre-commit hook (with husky https://typicode.github.io/husky and pretty-quick https://github.com/azz/pretty-quick)

#### What should we test?
It's a dev one. 

This added pre-commit hook that should work as following (you should run `yarn`):
 - Create or modify a file that is not ignored by `.prettierignore`, such as `app/webpacker/controllers/toggle_controller.js` by introducing a formatting _error_, such as: 
```diff 
diff --git a/app/webpacker/controllers/toggle_controller.js b/app/webpacker/controllers/toggle_controller.js
index de87ad095..15823eadb 100644
--- a/app/webpacker/controllers/toggle_controller.js
+++ b/app/webpacker/controllers/toggle_controller.js
@@ -6,7 +6,7 @@ export default class extends Controller {
   toggle(event) {
     const input = event.currentTarget;
     this.contentTargets.forEach((t) => {
-      t.style.display = input.dataset.toggleShow === "true" ? "block" : "none";
+        t.style.display = input.dataset.toggleShow === "true" ? "block" : "none";
     });
   }
```
 - Check that you could spot the error with `yarn prettier --check .` that output: 

```
Checking formatting...
[warn] app/webpacker/controllers/toggle_controller.js
[warn] Code style issues found in the above file(s). Forgot to run Prettier?
error Command failed with exit code 1.
```

 - Commit the modified file. Just before commiting (that's what we called a pre-commit hook), the following command should be run `yarn pretty-quick --staged` that format all files that are staged. Then the file should be well formatted, automatically. 

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

**Comment by jibees** [comment]
@mkllnk 
I can't make gh action  reviewdog/prettier working. 
I'm facing an issue which i do not understand:

```
📝 Running prettier with reviewdog 🐶 ...
  reviewdog: This GitHub token doesn't have write permission of Review API [1], 
  reviewdog: Reporting results for "prettier"
  so reviewdog will report results via logging command [2] and create annotations similar to
  reviewdog: No results found for "prettier". 16 results found outside diff.
  github-pr-check reporter as a fallback.
  [1]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target, 
  [2]: https://help.github.com/en/actions/automating-your-workflow-with-github-actions/development-tools-for-github-actions#logging-commands
```

Reference: https://github.com/openfoodfoundation/openfoodnetwork/runs/7497301810?check_suite_focus=true

If you could have a look? Thanks!

**Comment by mkllnk** [comment]
I'm not that familiar with that problem either. Did you try adding the token like in the docs?
https://github.com/EPMatt/reviewdog-action-prettier#usage

**Comment by jibees** [comment]
> I'm not that familiar with that problem either. Did you try adding the token like in the docs? https://github.com/EPMatt/reviewdog-action-prettier#usage

Yeap, I tried this one too (that was my first thought also)

**Review by mkllnk — APPROVED** [review]
Great, that's something to work with for now.

**Comment by mkllnk** [comment]
## Testing notes

My node version v12.18.2 was too old for husky. So I updated:
```
$ nvm install --lts
Installing latest LTS version.
Downloading and installing node v16.16.0...
Downloading https://nodejs.org/dist/v16.16.0/node-v16.16.0-linux-x64.tar.xz...
############################################################################################################################################################################################################ 100.0%
Computing checksum with sha256sum
Checksums matched!
Now using node v16.16.0 (npm v8.11.0)
```

Now again:

```
$ yarn
...
$ husky install
husky - Git hooks installed
```
That installed the Git hook already, I didn't need to run `yarn prepare`.

I introduced a formatting issue and ran `yarn prettier --check .` which raises an error. :heavy_check_mark: 

Then I try to commit and it just does it. The file is automatically formatted and I get an empty commit. Hm, I would prefer the git hook to prevent the commit and tell me what's wrong instead. Is that just my preference?

**Comment by jibees** [comment]
> The file is automatically formatted and I get an empty commit. Hm, I would prefer the git hook to prevent the commit and tell me what's wrong instead. Is that just my preference?

I suppose that the case where your file is finally empty should be relatively rare. 

We can also add `--check` to the command, to finally have `yarn pretty-quick --staged --check`, and then result is:

```
➜  openfoodnetwork git:(add-prettier) ✗ git commit -m"couco"
yarn run v1.22.19
$ pretty-quick --staged --check
🔍  Finding changed files since git revision db4b65ec9.
🎯  Found 1 changed file.
⛔️  Check failed: app/webpacker/controllers/example_controller.js
✗ Code style issues found in the above file(s). Forgot to run Prettier?
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
husky - pre-commit hook exited with code 1 (error)
```

**Comment by jibees** [comment]
> That installed the Git hook already, I didn't need to run yarn prepare

Yep, I just realized that `yarn` install the hook, nice! 👍

File: .github/workflows/linters.yml [hunk 1]
```diff
@@ -16,3 +16,16 @@ jobs:
...
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: prettier
+        uses: EPMatt/reviewdog-action-prettier@v1
+        with:
+          github_token: ${{ secrets.github_token }}
+          reporter: github-pr-check
+          level: error
+          fail_on_error: true
```
[the remark this answers is not in the document]
Comment (jibees) [reply .github/workflows/linters.yml:31]: Yes, or maybe open another PR?

**Comment by mkllnk** [comment]
I added the `--check` parameter.

**Pull request merged**